clients/scheduler/src/components/time-slot-select/ui.tsx

```diff
@@ -22,10 +22,20 @@ type SelectTimeSlotType = {
 }
 
 export const TimeSlotUi = ({ timeSlots }: UiPropsType) => {
-  const { colors, setTimeSlot, loading, shadowRoot, resetTimeSlot, providers } =
+  const { colors, date, timeSlot, setTimeSlot, preloadBooking, loading, shadowRoot, resetTimeSlot, providers } =
     useAppContext()
   const [popoverOpen, setPopoverOpen] = useState(false)
 
+  useEffect(() => {
+    if (preloadBooking && preloadBooking.start !== timeSlot.start) {
```

Comment:
will `timeSlot` always be defined here?

Reply:
Yes it is assigned default values in the `app-context.ts` file